src/modules/achievements/sagas.js

```diff
@@ -0,0 +1,135 @@
+import api from 'services/api'
+import stream from 'services/stream'
+import S from 'modules/selectors'
+import T from 'modules/types'
+import { eventChannel } from 'redux-saga'
+import blockchains from 'configurables/blockchains'
+
+const create = function * ({ link, title }) {
+  try {
+    const primaryBlockchainKey = blockchains.primary.key
+    const userAddress = yield select(S.wallets.primaryAddress)
+    yield call(api.createAchievement(primaryBlockchainKey), {
+      address: userAddress,
+      link,
+      name: yield select(S.login.userName),
+      title,
+      token: yield select(S.login.userAccessToken),
+      user: yield select(S.login.userID)
+    })
+    yield put(ownA.create.succeeded())
+  } catch (error) {
+    yield put(ownA.create.errored({ error }))
+  }
+}
+
+const confirm = function * (payload) {
+  try {
+    yield call(api.confirmAchievement, payload)
+    yield put(ownA.confirm.succeeded())
+  } catch (error) {
+    yield put(ownA.confirm.errored({ error }))
+  }
+}
+
+const fetch = function * ({ page }) {
+  try {
+    const { results: items, hasMore } = yield call(stream.fetchData, 'achievement_aggregated', 'common', page)
+    yield put(ownA.fetch.succeeded({ list: items, hasMore }))
+  } catch (error) {
+    yield put(ownA.fetch.errored({ error }))
+  }
+}
+
+const suscribe = function * () {
+  let callbackObj = {}
+  const channel = eventChannel(emitter => {
+    callbackObj.call = emitter
+    return () => {}
+  })
+  try {
+    yield call(stream.suscribeWithCallBacks, 'achievement_aggregated', 'common', callbackObj.call)
+    yield put(ownA.suscribe.succeeded())
+    while (true) {
+      yield take(channel)
+      yield call(fetch)
+      yield put(ownA.received())
+    }
+  } catch (error) {
+    yield put(ownA.suscribe.errored({ error }))
+  }
+}
+
+const fetchUserAchievements = function * ({ page }) {
+  try {
+    const userAddress = yield select(S.wallets.primaryAddress)
+    const { results: items, hasMore } = yield call(stream.fetchData, 'achievement_aggregated', userAddress, page)
+    yield put(ownA.fetchUser.succeeded({ list: items, hasMore }))
+  } catch (error) {
+    console
+    yield put(ownA.fetchUser.errored({ error }))
+  }
+}
+
+const suscribeUserAchievements = function * ({ userAddress }) {
+  let callbackObj = {}
+  const channel = eventChannel(emitter => {
+    callbackObj.call = emitter
+    return () => {}
+  })
+  try {
+    yield call(stream.suscribeWithCallBacks, 'achievement_aggregated', userAddress, callbackObj.call)
+    yield put(ownA.suscribe.succeeded())
+    while (true) {
+      yield take(channel)
+      yield call(fetchUserAchievements, { userAddress })
+      yield put(ownA.receivedUser())
+    }
+  } catch (error) {
+    yield put(ownA.suscribe.errored({ error }))
+  }
+}
+
+const support = function * ({ amount, blockchainKey, creatorAddress, fees, link }) {
+  try {
+    const { address, encodedData } = yield call(api.encodeSupport(blockchainKey), { link })
```

Comment:
Here we have a call to API before user send a transaction.

Note: why do we need this
Although we can prepare, sign and broadcast transaction on client side, some blockchains require heavy libraries. Therefore, in order to load webpage fast, here we can offload client and move heavy logic to backend service. Later we are planning to implement light SDK to generate transactions for all blockchains on client side, but for now, we keep an opportunity to request a backend to prepare transactions for some blockchains.

Suggestion: two steps to support other blockchains
1) Change name `encodeSupport` to `prepareSupport` in this place and other places in the code
2) Pass address, amount, fee rate, and wallet address to `prepareSupport`

Result: how it works with the lightweight client
1) Backend service prepare raw transaction (if specific blockchain requires heavy computations)
2) Client verifies transaction and sign the transaction with user's private key
3) Client sends a signed transaction to the backend that will broadcast the transaction to available node

Reply:
Note again, that this approach is not secure because backend can be compromised, but it is simpler to user for testing, and we will fix it next release.

Reply:
I did not understood if you want to change it right now or not?

Reply:
Yes, I want to use loom instead of qtum right now, and the simplest way to do it, is to change encodeSupport to prepareSupport